CONTRIBUTING.md

```diff
@@ -1,65 +1,189 @@
+In the past this was organized using Redmine at: [dev.exiv2.org](http://dev.exiv2.org). As of April 2017 the Exiv2 project has moved from svn and Redmine to GitHub and information, such
+as, Wikis etc., will gradually be moved from Redmine to GitHub. User requests on the Redmine forums and the Redmine issue tracking system will be serviced until Exiv2 0.27 is released. 
+After 0.27 is released the Redmine site will be read-only and all feature requests, bug reports etc., must be performed on GitHub. It is, however, preferred to use GitHub for all new 
+feature requests and bug reports to Exiv2.
 
-3. Make sure the tests pass:
-```
-make tests          # Application tests
-./bin/unit_tests    # Unit tests
-```
+# 2. Recommended Workflow for Contributing Code #
+
+Here we describe the recommended code contribution workflow after switching to GitHub. Code contributions is performed by using *pull requests* (PR) on GitHub. 
+For this to work one first needs to [create a user account on GitHub](https://help.github.com/articles/signing-up-for-a-new-github-account/) if one do not 
+already have one. A PR should preferable contain only one new feature/bug fix etc. Since it is not uncommon to work on several PRs at the same time 
+it is recommended to create a new _branch_ for each PR. In this way PRs can easily be separated and the review and merge process becomes cleaner.
+As a rule-of-thumb:
+
+- PRs should be kept at a manageable size. Try to focus in just one goal per PR. If you find yourself doing several things in a PR that were not expected, 
+then try to split the different tasks in different PRs.
+- Commits should always change a *single* logical unit so that cherry-picking & reverting is simple.
+- Commit messages should be as informative and concise as possible. The first line of the commit message should be < 80 characters and 
+  describe the commit briefly. If the 80 characters are too short for a summary, then consider splitting the commit. Alternatively, below the short summary 
+  add one blank line and then a more detailed explanation if required.
+
+See the [GIT_GUIDELINES.md](GIT_GUIDELINES.md) file for a more detailed description on the git workflow. 
+
+Below we outline the recommended steps in the code contribution workflow. We use `your-username` to refer to your username on GitHub, `exiv2_upstream` is used when we 
+set the upstream remote repository for Exiv2 (we could have picked any name except `origin` and `master` [TODO verify this]), and we use the name `my_first_pr_branch` 
+for the branch that we create (eg., the branch name should reflect the code change being made).
+
+Once you have a GitHub login:
+
+1. Fork the Exiv2 git repository to your GitHub account by pressing the _Fork_ button at: [https://github.com/Exiv2/exiv2](https://github.com/Exiv2/exiv2) 
```

Comment:
Exiv2 team members should work either in their fork or create a branch in the main repository to allow collaboration when applicable.

Reply:
Dan, to work on a branch on the main repo don't  one need write (push) rights to that repo?

Reply:
Yes, you need to have push rights. But if you have them, it can be beneficial to work in a branch in the main repository, since others can push there too. In your fork only you can, which makes it awkward if others want to add commits to a PR too.

Reply:
I will write a paragraph that explains that there are two methods but that we first will treat the "fork" method since I think it is important not to move too fast for beginners. It will anyway take a while before one gets push rights to the main repo, right?

Reply:
I think a remark, that people with push rights to the main repo should create branches in the exiv2 repo (when applicable), is enough. That way we have it documented also for team members and one-time contributors or newcommers don't get too confused by that.

Yes, it usually takes a while before you get push access.